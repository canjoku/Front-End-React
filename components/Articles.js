import { Component } from 'react'

import PostCard from '../components/PostCard'


class Articles extends Component {

  render () {
    const { posts } = this.props
    return (
      <div className='c-articles'>
        <h1 className='c-articles__primary-heading'>Articles</h1>
        <div className='c-articles__posts-container'>
          { posts.map((post, index) => {
            return(
              <div key={index} className='c-articles__posts-container-item'>
                <PostCard post={post}/>
              </div>
            )
          })}
        </div>
      </div>
    )
  }
}

export default Articles