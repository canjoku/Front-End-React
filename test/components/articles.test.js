import Articles from '../../components/Articles'


test('<Articles /> renders correctly', () => {
  const post =
  [{
    title: "title of first post",
    body: "body of first post",
    createdAt: "18-01-2019"
  }]

  const component = shallow(<Articles posts={ post }/>)
  expect(component).toMatchSnapshot()
})
