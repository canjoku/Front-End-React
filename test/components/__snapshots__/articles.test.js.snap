// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Articles /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Articles
    posts={
      Array [
        Object {
          "body": "body of first post",
          "createdAt": "18-01-2019",
          "title": "title of first post",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="c-articles__primary-heading"
        >
          Articles
        </h1>,
        <div
          className="c-articles__posts-container"
        >
          <div
            className="c-articles__posts-container-item"
          >
            <PostCard
              post={
                Object {
                  "body": "body of first post",
                  "createdAt": "18-01-2019",
                  "title": "title of first post",
                }
              }
            />
          </div>
        </div>,
      ],
      "className": "c-articles",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Articles",
          "className": "c-articles__primary-heading",
        },
        "ref": null,
        "rendered": "Articles",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="c-articles__posts-container-item"
            >
              <PostCard
                post={
                  Object {
                    "body": "body of first post",
                    "createdAt": "18-01-2019",
                    "title": "title of first post",
                  }
                }
              />
            </div>,
          ],
          "className": "c-articles__posts-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": <PostCard
                post={
                  Object {
                    "body": "body of first post",
                    "createdAt": "18-01-2019",
                    "title": "title of first post",
                  }
                }
              />,
              "className": "c-articles__posts-container-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "post": Object {
                  "body": "body of first post",
                  "createdAt": "18-01-2019",
                  "title": "title of first post",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="c-articles__primary-heading"
          >
            Articles
          </h1>,
          <div
            className="c-articles__posts-container"
          >
            <div
              className="c-articles__posts-container-item"
            >
              <PostCard
                post={
                  Object {
                    "body": "body of first post",
                    "createdAt": "18-01-2019",
                    "title": "title of first post",
                  }
                }
              />
            </div>
          </div>,
        ],
        "className": "c-articles",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Articles",
            "className": "c-articles__primary-heading",
          },
          "ref": null,
          "rendered": "Articles",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="c-articles__posts-container-item"
              >
                <PostCard
                  post={
                    Object {
                      "body": "body of first post",
                      "createdAt": "18-01-2019",
                      "title": "title of first post",
                    }
                  }
                />
              </div>,
            ],
            "className": "c-articles__posts-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": <PostCard
                  post={
                    Object {
                      "body": "body of first post",
                      "createdAt": "18-01-2019",
                      "title": "title of first post",
                    }
                  }
                />,
                "className": "c-articles__posts-container-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "post": Object {
                    "body": "body of first post",
                    "createdAt": "18-01-2019",
                    "title": "title of first post",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
